test/e2e/e2e.go

```diff
@@ -1309,10 +1309,12 @@ spec:
 		framework.RemoveLabelOffNode(f.ClientSet, egress1Node.name, "k8s.ovn.org/egress-assignable")
 
 		ginkgo.By("9. Check that the status is of length one and that it is assigned to egress2Node")
-		statuses = verifyEgressIPStatusLengthEquals(1)
 		// There is sometimes a slight delay for the EIP fail over to happen,
 		// so let's use the pollimmediate struct to check if eventually egress2Node becomes the egress node
-		err = wait.PollImmediate(retryInterval, retryTimeout, func() (bool, error) { return statuses[0].Node == egress2Node.name, nil })
+		err = wait.PollImmediate(retryInterval, retryTimeout, func() (bool, error) { 
+			statuses := getEgressIPStatusItems()
```

Comment:
I think we should add a length check to avoid it crashing if we break something.

Reply:
hmm should we? I though during re-assignment we'd usually just patch the nodeName and length wouldn't change, but you are right.. better safe than sorry. Let me fix this.